presto-main/src/main/java/com/facebook/presto/execution/NodeTaskMap.java

```diff
@@ -108,87 +146,161 @@ private void addTask(RemoteTask task)
-            private final AtomicInteger localPartitionedSplitCount = new AtomicInteger();
+            private final AtomicLong totalValue;
+            private final AtomicLong value = new AtomicLong();
 
-            public TaskPartitionedSplitCountTracker(TaskId taskId)
+            public TaskStatsTracker(String stat, TaskId taskId, AtomicLong totalValue)
             {
+                this.stat = requireNonNull(stat, "stat is null");
                 this.taskId = requireNonNull(taskId, "taskId is null");
+                this.totalValue = requireNonNull(totalValue, "totalValue is null");
             }
 
-            public synchronized void setPartitionedSplitCount(int partitionedSplitCount)
+            public synchronized void setValue(long value)
             {
-                if (partitionedSplitCount < 0) {
-                    int oldValue = localPartitionedSplitCount.getAndSet(0);
-                    nodeTotalPartitionedSplitCount.addAndGet(-oldValue);
-                    throw new IllegalArgumentException("partitionedSplitCount is negative");
+                if (value < 0) {
+                    long oldValue = this.value.getAndSet(0L);
+                    totalValue.addAndGet(-oldValue);
+                    throw new IllegalArgumentException(stat + " is negative");
                 }
 
-                int oldValue = localPartitionedSplitCount.getAndSet(partitionedSplitCount);
-                nodeTotalPartitionedSplitCount.addAndGet(partitionedSplitCount - oldValue);
+                long oldValue = this.value.getAndSet(value);
+                totalValue.addAndGet(value - oldValue);
             }
 
             public void cleanup()
             {
-                int leakedSplits = localPartitionedSplitCount.getAndSet(0);
-                if (leakedSplits == 0) {
+                long leakedValues = value.getAndSet(0);
+                if (leakedValues == 0) {
                     return;
                 }
 
-                log.error("BUG! %s for %s leaked with %s partitioned splits.  Cleaning up so server can continue to function.",
+                log.error("BUG! %s for %s leaked with %s %s.  Cleaning up so server can continue to function.",
                         getClass().getName(),
                         taskId,
-                        leakedSplits);
+                        leakedValues,
+                        stat);
 
-                nodeTotalPartitionedSplitCount.addAndGet(-leakedSplits);
+                totalValue.addAndGet(-leakedValues);
             }
 
             @Override
             public String toString()
             {
                 return toStringHelper(this)
                         .add("taskId", taskId)
-                        .add("splits", localPartitionedSplitCount)
+                        .add(stat, value)
                         .toString();
             }
         }
+
+        @ThreadSafe
+        private class TaskCumulativeStatsTracker
```

Comment:
```suggestion
        private class CumulativeTaskStatsTracker
```

Reply:
Can you comment what this class is trying to do?  The class name doesn't seem to describe it well.